plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/cleaner/DockerAbandonedResourcesCleaner.java

```diff
@@ -96,23 +96,27 @@ public void run() {
      * Cleans up CHE docker containers which don't tracked by API any more.
      */
     @VisibleForTesting
-    void cleanContainers() {
+    void cleanContainers(@Nullable @Named("che.server_id") String cheServerId) {
         List<String> activeContainers = new ArrayList<>();
         try {
             for (ContainerListEntry container : dockerConnector.listContainers()) {
                 String containerName = container.getNames()[0];
                 Optional<ContainerNameInfo> optional = nameGenerator.parse(containerName);
-                if (optional.isPresent()) {
-                    try {
-                        // container is orphaned if not found exception is thrown
-                        environmentEngine.getMachine(optional.get().getWorkspaceId(),
-                                                     optional.get().getMachineId());
-                        activeContainers.add(containerName);
-                    } catch (NotFoundException e) {
-                        cleanUpContainer(container);
-                    } catch (Exception e) {
-                        LOG.error(format("Failed to check activity for container with name '%s'. Cause: %s",
-                                         containerName, e.getLocalizedMessage()), e);
+                if (optional.isPresent() ) {
+                    if(optional.get().getServerId().equals(cheServerId)){
+                        try {
+                            // container is orphaned if not found exception is thrown
+                            if(optional.get().getServerId().equals("") || ){
```

Comment:
Assumption that it is empty string violates encapsulation principle. I suggest to not expose serverId identifier in ContainerNameInfo at all - this is information not needed for other classes. So in that case if server ID doesn't match `DockerContainerNameGenerator.parse(containerName)` can return empty optional and container is treated as non-matching.